modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java

```diff
@@ -117,6 +117,58 @@ public void testCreateServiceProviderRequest() throws IOException {
 
     }
 
+    @Test(alwaysRun = true, groups = "wso2.is", priority = 1, description = "Create a service provider with optional " +
+            "client metadata values successfully")
+    public void testCreateServiceProviderRequestWithOptionalAttributes() throws IOException {
+        HttpPost request = new HttpPost(getPath());
+        setRequestHeaders(request);
+
+        JSONArray grantTypes = new JSONArray();
+        grantTypes.add(OAuthDCRMConstants.GRANT_TYPE_AUTHORIZATION_CODE);
+        grantTypes.add(OAuthDCRMConstants.GRANT_TYPE_IMPLICIT);
+
+        JSONArray redirectURI = new JSONArray();
+        redirectURI.add(OAuthDCRMConstants.REDIRECT_URI);
+
+        JSONArray audience = new JSONArray();
+        audience.add(OAuthDCRMConstants.AUD_VALUE);
+
+        JSONObject dcrRequest = new JSONObject();
+        dcrRequest.put(OAuthDCRMConstants.CLIENT_NAME, OAuthDCRMConstants.APPLICATION_NAME_OPTIONAL_ATTRIBUTES);
+        dcrRequest.put(OAuthDCRMConstants.GRANT_TYPES, grantTypes);
+        dcrRequest.put(OAuthDCRMConstants.REDIRECT_URIS, redirectURI);
+        dcrRequest.put(OAuthDCRMConstants.AUD, audience);
+        dcrRequest.put(OAuthDCRMConstants.SOFTWARE_ID, OAuthDCRMConstants.SOFTWARE_ID_VALUE);
+        dcrRequest.put(OAuthDCRMConstants.TOKEN_ENDPOINT_AUTH_METHOD, OAuthDCRMConstants.TOKEN_ENDPOINT_AUTH_METHOD_VALUE);
+        dcrRequest.put(OAuthDCRMConstants.ID_TOKEN_ENCRYPTED_RESPONSE_ALG,
+                OAuthDCRMConstants.ID_TOKEN_ENCRYPTED_RESPONSE_ALG_VALUE);
+        dcrRequest.put(OAuthDCRMConstants.ID_TOKEN_ENCRYPTED_RESPONSE_ENC,
+                OAuthDCRMConstants.ID_TOKEN_ENCRYPTED_RESPONSE_ENC_VALUE);
+
+        StringEntity entity = new StringEntity(dcrRequest.toJSONString());
+
+        request.setEntity(entity);
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 201, "Service Provider " +
+                "has not been created successfully");
```

Comment:
is the message provided here correct?

Reply:
This message will be displayed if the service provider is not created successfully. (if status code returned is not equal to 201)